internal/syncer/git_commit_stats.go

```diff
@@ -150,7 +150,7 @@ func (w *worker) handleGitCommitStats(ctx context.Context, j *db.DequeueSyncJobR
 		}
 		defer func() {
 			if err := diff.Free(); err != nil {
-				w.logger.Err(err).Msgf("error freeing diff")
+				w.logger.Warn().AnErr("Error", err).Msgf("error freeing diff")
```

Comment:
I think this one should be an actual error vs a warning - we'll want to know why we can't call `.Free()`